lib/rewardable/Limiter.cpp

```diff
@@ -33,6 +35,37 @@ Rewardable::Limiter::Limiter()
 
 Rewardable::Limiter::~Limiter() = default;
 
+bool Rewardable::Limiter::empty() const
+{
+	if(dayOfWeek != 0
+	   || daysPassed != 0
+	   || heroLevel > 0
+	   || heroExperience > 0
+	   || manaPoints > 0
+	   || manaPercentage > 0
```

Comment:
```suggestion
	   || manaPercentage >= 0
```
manaPercentage == 0 is "take all mana"

Reply:
for reward yes, but for limiter no. It checks if player has mana more than x%